File: internal/content/testdata/bench/rust/nc-car-fleet.rs
fn car_fleet(target: i64, position: Vec<i64>, speed: Vec<i64>) -> i64 {
    let mut pairs: Vec<(i64, i64)> = position
        .iter()
        .cloned()
        .zip(speed.iter().cloned())
        .collect();
    // sort by position descending (reverse of ascending zip sort)
    pairs.sort_by(|a, b| b.cmp(a));
    let mut stack: Vec<f64> = Vec::new();
    for &(pos, spd) in pairs.iter() {
        let time = (target - pos) as f64 / spd as f64;
        if stack.is_empty() || time > *stack.last().unwrap() {
            stack.push(time);
        }
    }
    stack.len() as i64
}

File: internal/content/testdata/bench/typescript/nc-car-fleet.ts
function car_fleet(target: number, position: number[], speed: number[]): number {
    const pairs: [number, number][] = position.map((p, i) => [p, speed[i]]);
    pairs.sort((a, b) => b[0] - a[0]);
    const stack: number[] = [];
    for (const [pos, spd] of pairs) {
        const time = (target - pos) / spd;
        if (stack.length === 0 || time > stack[stack.length - 1]) {
            stack.push(time);
        }
    }
    return stack.length;
}
